web/frontend/src/components/camera.vue

```diff
@@ -26,13 +28,41 @@ const emit = defineEmits<Emits>();
 
 let pcdExpanded = $ref(false);
 let pointcloud = $ref<Uint8Array | undefined>();
+let camera = $ref(false);
 
-const camera = ref(false);
 const selectedValue = ref('live');
 
+const initStreamState = () => {
+  cameraStreamStates.set(props.cameraName, false);
+};
+
+const viewCamera = async (isOn: boolean) => {
+  if (isOn) {
+    try {
+      // only add stream if not already active
+      if (!baseStreamStates.get(props.cameraName) && !cameraStreamStates.get(props.cameraName)) {
```

Comment:
two small recs: change `.get()` to `.has()` (since that'll return a boolean) and move this if statement outside the `try {} catch {}` since only awaiting the promise can possibly throw an error

Reply:
As discussed, this has to be get as I'm not removing items from map.